File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/stored/system/IMediaQueryCursorProvider.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.stored.system;

import android.database.Cursor;

import com.lasthopesoftware.bluewater.client.library.items.media.files.ServiceFile;
import com.lasthopesoftware.messenger.promises.Promise;

/**
 * Created by david on 6/13/16.
 */
public interface IMediaQueryCursorProvider {
	Promise<Cursor> getMediaQueryCursor(ServiceFile serviceFile);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/system/IMediaQueryCursorProvider.kt
package com.lasthopesoftware.bluewater.client.stored.library.items.files.system

import android.database.Cursor
import com.lasthopesoftware.bluewater.client.browsing.files.ServiceFile
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.namehillsoftware.handoff.promises.Promise

/**
 * Created by david on 6/13/16.
 */
interface IMediaQueryCursorProvider {
	fun getMediaQueryCursor(libraryId: LibraryId, serviceFile: ServiceFile): Promise<Cursor?>
}
